src/metric/definition.go

```diff
@@ -1595,3 +1597,67 @@ func metricSetTypeGuesserWithCustomGroup(group string) definition.GuessFunc {
 		return definition.K8sMetricSetTypeGuesser(group) //nolint: wrapcheck
 	}
 }
+
+// error checks.
+var (
+	errFetchedValueTypeCheck = fmt.Errorf("fetchedValue must be of type float64")
+	errCPULimitTypeCheck     = fmt.Errorf("cpuLimit must be of type float64")
+	errGroupLabelCheck       = fmt.Errorf("group label not found")
+	errEntityCheck           = fmt.Errorf("entity Id not found")
+	errHighCPUUsedCores      = fmt.Errorf("impossibly high value received from kubelet for cpuUsedCoresVal")
+)
+
+// filterCPUUsedCores checks for the correctness of the container metric cpuUsedCores returned by kubelet.
+// cpuUsedCores a.k.a `usageNanoCores` value is set by cAdvisor and is returned by kubelet stats summary endpoint.
+// There is an active bug where the metric value is sometimes impossibly high https://github.com/kubernetes/kubernetes/issues/114057.
+// The cpuUsedCores along with cpuLimitCores is typically used to plot `cpuCoresUtilization` on the UI where cpuCoresUtilization = (cpuUsedCores/cpuLimitCores) * 100.
+// cpuUsedCores has been observed to be absurd even when cpuUsedCores >  cpuLimitCores * 100.
+//
+//nolint:nolintlint,ireturn
+func filterCPUUsedCores(fetchedValue definition.FetchedValue, groupLabel, entityID string, groups definition.RawGroups) (definition.FilteredValue, error) {
+	// type assertion check
+	val, ok := fetchedValue.(float64)
+	if !ok {
+		return nil, errFetchedValueTypeCheck
+	}
+
+	// fetch raw cpuLimitCores value
+	group, ok := groups[groupLabel]
+	if !ok {
+		return nil, errGroupLabelCheck
+	}
+
+	entity, ok := group[entityID]
+	if !ok {
+		return nil, errEntityCheck
+	}
+
+	value, ok := entity["cpuLimitCores"]
+	if !ok {
+		// there is likely no CPU limit set for the container which means we have to assume a reasonable value
+		// since there is no way to know the max cpu cores for the current node, use default max of 96 cores supported by most cloud providers
+		// a higher value wouldn't hurt our calculation as the cpuUsedCores value will be a super high number
+		log.StandardLogger().Warnf("cpuLimitCores metric not available. using default max 96 cores")
+		value = 96000 // 9600m k8s cpu unit
```

Comment:
Can you fix the typo in the comment? It is confusing to see both a constant of 96000 and the comment having a different value